### ISSUE TITLE
Question: How to disable -O0 default compile flag?

### ISSUE THREAD

**Issue by leigh123linux** [MusicPlayerDaemon/MPD#1720]
I would like to set my own flags, how do I remove the -O0 flag?
I have looked at the meson build file and get see where it is set.

``
[2/682] ccache c++ -Isrc/util/libutil.a.p -Isrc/util -I../src/util -Isrc -I../src -I. -I.. -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -O0 -ffast-math -ftree-vectorize -Wcast-qual -Wdouble-promotion -Wmissing-declarations -Wshadow -Wunused -Wvla -Wwrite-strings -Wno-stringop-overflow -fno-threadsafe-statics -fmerge-all-constants -Wcomma-subscript -Wextra-semi -Wmismatched-tags -Woverloaded-virtual -Wsign-promo -Wvolatile -Wvirtual-inheritance -Wno-non-virtual-dtor -Wsuggest-override -fvisibility=hidden -ffunction-sections -fdata-sections -D_GNU_SOURCE -fPIC -MD -MQ src/util/libutil.a.p/Exception.cxx.o -MF src/util/libutil.a.p/Exception.cxx.o.d -o src/util/libutil.a.p/Exception.cxx.o -c ../src/util/Exception.cxx
```

**Comment by MaxKellermann** [comment]
https://mesonbuild.com/Builtin-options.html#core-options

**Comment by leigh123linux** [comment]
I am using plain.

```
+ /usr/bin/meson --buildtype=plain --prefix=/usr --libdir=/usr/lib64 --libexecdir=/usr/libexec --bindir=/usr/bin --sbindir=/usr/sbin --includedir=/usr/include --datadir=/usr/share --mandir=/usr/share/man --infodir=/usr/share/info --localedir=/usr/share/locale --sysconfdir=/etc --localstatedir=/var --sharedstatedir=/var/lib --wrap-mode=nodownload --auto-features=enabled . redhat-linux-build -Dsystemd_system_unit_dir=/usr/lib/systemd/system -Dsystemd_user_unit_dir=/usr/lib/systemd/user -Dadplug=disabled -Dipv6=enabled -Dpipe=true -Ddocumentation=auto -Dsolaris_output=disabled -Dsndio=disabled -Dchromaprint=disabled -Dgme=disabled -Dmpcdec=disabled -Dshine=disabled -Dtremor=disabled
The Meson build system
Version: 0.62.2
Source dir: /builddir/build/BUILD/mpd-0.23.11
Build dir: /builddir/build/BUILD/mpd-0.23.11/redhat-linux-build
Build type: native build
```

**Comment by MaxKellermann** [comment]
This is a Meson problem, not a MPD problem. I don't know how to tell Meson not to use any optimization flag, because I have never felt the need to.

**Comment by leigh123linux** [comment]
The issue is fixed in meson-0.64.0 https://github.com/mesonbuild/meson/commit/a590cfde0cf719c637b75e4784be0c0ae60e3b1f
